### PR TITLE
feat(cfb): add load_cfb_teams for the espn_cfb_teams release

### PR DESCRIPTION
**Pull request by saiemgilani**
`espn_cfb_teams` has had no loader since it was published — callers had to hand-build the release URL. This adds one, with the 54-column schema the dataset actually ships (not the 37 it carried before the CFBD backport).

## What the dataset covers

Every ESPN CFB team per season, 2001–2026 — the whole NCAA tree, not just Division I. 2023 has **857 teams** against the 145 in group 80; the rest are D-II/D-III (including teams ESPN files directly under the Division II/III node) and the **parentless NAIA group 186**, which no tree walk reaches.

## Filtering

```python
teams = load_cfb_teams([2023])                              # 857 x 54
fbs   = teams.filter(pl.col("is_fbs"))                      # 145
real  = teams.filter(pl.col("is_fbs") & ~pl.col("is_exhibition"))  # 133
```

Use `is_fbs` rather than `division == "fbs"`: `is_fbs` is a non-null Boolean, whereas a null-division comparison yields null in polars and silently drops rows from a mask. `is_fbs & ~is_exhibition` gives the real program count — ESPN files bowl all-star squads inside the FBS group, and its own `isAllStar` flag catches only about a third of them. That combination independently reproduces CFBD's FBS set in 15 of 22 seasons.

## Column documentation

All 54 descriptions are new. Two families needed disambiguating after the CFBD backport merged both feeds into one row:

- ESPN's `venue_city` / `venue_indoor` / `venue_grass` describe the **stadium**; CFBD's `city` / `dome` / `grass` describe the **school**. They can disagree.
- `classification` is CFBD's independent second opinion on `division`, documented as something to inspect on disagreement rather than reconcile blindly.

## Verification

```
2023 -> (857, 54), team_id Int64
multi-season [2015, 2023, 2025] -> (2447, 54)
season floor -> SeasonNotFoundError below 2001
codegen tests: 158 passed, 11 skipped
ruff: clean · codegen --check: clean · uv.lock: untouched
```

## Note for the next loader addition

`sportsdataverse/parsed/cfb.py` re-exports loaders by importing them at runtime, so the **first** `generate.py` pass renders it before `cfb_loaders.py` has the new function — `--check` then reports `parsed/cfb.py` stale while the file isn't even modified. A second pass settles it. Worth knowing, since that is exactly the kind of drift that reddens CI after a locally-green run.

## Summary by Sourcery

Add a CFB team dataset loader backed by the `espn_cfb_teams` release and document its 54-column schema.

New Features:
- Add a `load_cfb_teams` loader for the `espn_cfb_teams` release, supporting single- and multi-season data retrieval in Polars or pandas.
- Expose the loader through the CFB API and generated endpoint metadata.

Enhancements:
- Define the complete 54-column team schema, including ESPN team, conference, venue, and CFBD cross-source attributes with documented distinctions.
- Handle unavailable seasons safely, enforce the minimum supported season, normalize team IDs to Int64, and union schemas across seasons.
- Move `espn_cfb_teams` from the additional-functions documentation into the dataset-loader documentation.

Documentation:
- Document the new loader, release asset, supported fields, types, and filtering guidance.

## Summary by CodeRabbit

* **New Features**
  * Added a college football teams loader with season-based data access.
  * Included team, conference, venue, school, classification, and location details.
  * Supports returning results in either Polars or pandas format.
  * Added validation for unsupported seasons and warnings for unavailable data.

* **Documentation**
  * Added usage guidance, schema details, and release-status information.
  * Updated reference tables to reflect the new loader.

### CODE REVIEW

File: sportsdataverse/cfb/cfb_loaders.py [hunk 1]
```diff
@@ -1028,6 +1029,106 @@ def load_cfb_team_info(seasons, return_as_pandas: bool = False):
...
+
+            load_cfb_teams(seasons=2024)
+    """
+    frames, missing = [], []
+    for season in _as_season_list(seasons):
+        if int(season) < 2001:
+            raise SeasonNotFoundError("season cannot be less than 2001")
+        df = _read_release_parquet(
+            f"https://github.com/sportsdataverse/sportsdataverse-data/releases/download/espn_cfb_teams/cfb_teams_{season}.parquet"
+        )
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/cfb/cfb_loaders.py:1116]: Tracked as #397 rather than fixed here — the point is legitimate, but it is not a change this PR can correctly make.

`_read_release_parquet()` is the shared transport for every release asset in the package: **226 call sites** across 8 loader modules. Routing this one loader around it would make it the only inconsistent one, and your own `Heavy lift` label plus the suggestion to update the *release-reader / code-generation path* acknowledges the real scope.

Two behaviours any replacement has to preserve, which is why it needs its own change and tests: the helper classifies a missing asset into `None` for the 404-safe loader contract while deliberately re-raising 403 and parse errors so they are not masked as "no data"; and it recovers from a pyo3 `PanicException` on R-producer assets carrying `arrow.r.vctrs` metadata by re-reading with field metadata stripped. `download()` returns a `requests.Response`, so serving a columnar read through it means buffering 50 MB+ assets into memory — a real behaviour change for the pbp tags.

#397 records the inconsistency, what a correct fix must preserve, and the alternative of scoping the CLAUDE.md wording so the guideline stops firing on every release-loader PR.
